File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stringstream ss(path);
        string tmp;
        vector<string> stk;
        while(getline(ss, tmp, '/')) {
            if(tmp == "..") {
                if(!stk.empty()) stk.pop_back();
            } else if(tmp != "." && !tmp.empty()) {
                stk.push_back(tmp);
            }
        }
        string res = "";
        for(auto s: stk) res += "/" + s;
        return ((res.empty()) ? "/" : res);
    }
};

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        paths = path.split('/')
        res = []
        for path in paths:
            if path and path != '.':
                if path == '..':
                    if res: res.pop()
                else:
                    res.append(path)
        return '/' + '/'.join(res)